jwst/extract_1d/extract.py

```diff
@@ -2983,7 +2983,7 @@ def do_extract1d(input_model, ref_dict, smoothing_length=None,
                     else:
                         wcs = input_model.meta.wcs
                     # There's no need to repeat this for each integration.
-                    if integ == 0:
+                    if integ <= 1:
```

Comment:
The reason for this change is due to the fact that in upstream code ``integrations`` is set to "[-1]" when the input data have a shape of 2 (as opposed to 3), so when executing this loop ``integ`` is equal to -1. The test for a value of zero was then causing this part of the if-block to get skipped, resulting the the pixel area values not getting populated. Now the pixel area is populated.

Reply:
This is JP-1005, GitHub #4040.  However, this whole section will be deleted by addressing JP-1106, GitHub #4207.